### DISCUSSION TITLE
Migrate database SqlServer and DotNet 8

### DISCUSSION THREAD

**Discussion by chogberg** [fluentmigrator/fluentmigrator#2019] [Q&A] [answered]
**Describe the question**
I'm quite new to DotNet, so there might be something that I missed or misunderstood. I started out with Fluent Migrator using the SQLite database, and it worked great. Then, I tried to switch to MSSQL but couldn't get the migrations running. I'm using the in-process runner as recommended. I have verified that everything is working by running the console runner, which migrates exactly as expected.

When running the in-process runner, the SQL queries that are supposed to be run are logged to the console instead. Nothing is created in the database, and running it multiple times gives the same output (which wouldn't be the case if the version table could be created and read). I do not get any indication that the migration failed.

This is my MSSQL part in program.cs:

```
private static IServiceCollection AddServices(IServiceCollection serviceCollection)
  {
        serviceCollection
            // Add common FluentMigrator services
            .AddFluentMigratorCore()
            .ConfigureRunner(runner => runner.AddSqlServer()
                // Set the connection string
                .WithGlobalConnectionString(_connectString)
                // Define the assembly containing the migrations
                .WithMigrationsIn(Assembly.GetExecutingAssembly())
            )
            // Enable logging to console in the FluentMigrator way
            .AddLogging(lb => lb.AddFluentMigratorConsole())
            .Configure<RunnerOptions>(opt => { opt.Tags = tags.ToArray(); });
        return serviceCollection;
    }
```
The connection string is the same as I used in the console runner, which works.

Example output when running the in-process runner:

```
 -------------------------------------------------------------------------------
3: CreateAppliancesTable migrating
-------------------------------------------------------------------------------
info: FluentMigrator.Runner.MigrationRunner[1002]
      => 0.0001653s
info: FluentMigrator.Runner.MigrationRunner[1000]
      3: CreateAppliancesTable migrating
CreateTable Appliances
info: FluentMigrator.Runner.MigrationRunner[1004]
      CreateTable Appliances
info: FluentMigrator.Runner.Processors.SqlServer.SqlServer2016Processor[1003]
      CREATE TABLE [dbo].[Appliances] ([idInt] BIGINT NOT NULL IDENTITY(1,1), [Id] UNIQUEIDENTIFIER NOT NULL, [Name] NVARCHAR(255) NOT NULL, [Image] VARBINARY(8000), [Textfield] NVARCHAR(255) NOT NULL, CONSTRAINT [PK_Appliances] PRIMARY KEY ([idInt]))
info: FluentMigrator.Runner.MigrationRunner[1002]
      => 0.003569s
CreateConstraint Unique_Appliances
info: FluentMigrator.Runner.MigrationRunner[1004]
      CreateConstraint Unique_Appliances
info: FluentMigrator.Runner.Processors.SqlServer.SqlServer2016Processor[1003]
      ALTER TABLE [dbo].[Appliances] ADD CONSTRAINT [Unique_Appliances] UNIQUE ([Id])
3: CreateAppliancesTable migrated
info: FluentMigrator.Runner.MigrationRunner[1002]
      => 0.001378s
info: FluentMigrator.Runner.Processors.SqlServer.SqlServer2016Processor[1003]
      INSERT INTO [dbo].[VersionInfo] ([Version], [AppliedOn], [Description]) VALUES (3, '2024-08-02T17:42:01', N'CreateAppliancesTable')
info: FluentMigrator.Runner.MigrationRunner[1004]
      3: CreateAppliancesTable migrated
info: FluentMigrator.Runner.MigrationRunner[1002]
      => 0.0014135s
```

**Documentation Pages You've Read So Far**
I have googled and read the documentation quite thoroughly but to no avail. Both when I got the Sqlite migrations running and trying to get MSSQL going.

**Expected benefit**
It would be great to get some help getting the migrations going. Maybe I could help clarify the documentation if something is missing or could be clearer.

**Information (please complete the following information):**
 - OS: Docker container running Debian 12 
 - Platform: Dotnet 8.0.0in a console application
 - FluentMigrator version: 5.2
 - FluentMigrator runner:  "in-process runner" 
 - Database Management System SQLServer
 - Database Management System Version: Docker image mcr.microsoft.com/azure-sql-edge

**Additional context**
I used this guide as a template (https://code-maze.com/dapper-migrations-fluentmigrator-aspnetcore/) when I could not get it going. It did not help, but right now, the implementation is kind of the same (the creation of the database is done manually right now)

Any pointers would be great since I'm out of ideas!

Regards
Chogberg

**Comment by chogberg** [comment]
Maybe this should be moved to discussions instead since I'm not sure if this is an issue with FluentMigrator or with me not being able to figure it out :-(

**Comment by jzabroski** [comment]
Sorry for the delay getting to your question. I had a newborn child recently.

I am a little confused by your question since you mentioned MySQL in addition to SQL Server (MSSQL) and SQLite.

> When running the in-process runner, the SQL queries that are supposed to be run are logged to the console instead.

It sounds like you are running in Preview mode.

**Comment by chogberg** [comment]
No worries! Im greatful that you do this at all!

Unfortunately, the MySQL was a typo. Sorry for that. I still haven't figured it out. I fully agree that it looks like preview mode, but the change from SQLite to MSSQL was just changing the runner, so I was kind of baffled when it did not work.

I agree that it looks like preview mode, but since I have no clue how to configure preview mode and the Sqlite setup was straightforward, so I'm kind of lost. Something in my config is probably strange, but I'm not familiar with fluent migrator, dotnet, or MSSQL to figure out where I have gone wrong.

**Comment by jzabroski** [comment]
if you can upload a full sample i can look.

I'm actually working on samples for each db. For selfish reasons of regression testing. I have sql server in my local sample but it's mixed in with a bigger pr I'm working on

**Comment by chogberg** [comment]
Sorry for my slow response. I've been quite busy at work! Is it possible to send it to you? I'm not sure if there is any sensitive information in it, so I'd rather not upload it publicly.

**Comment by jzabroski** [comment]
```<my first name>@simmaculate.com```

**Comment by dmshapoval** [comment, marked as answer]
@chogberg Faced the same behavior for Postgres and AspNet.Core 8. After debugging found that there was a typo in  configs and the connection string in `WithGlobalConnectionString(connectString)`  was null. This has led to ConnectionlessProcessor which shows all  sql commands but doesn't commit transaction. Maybe this help.

**Comment by PhenX** [comment]
I think it may be due to #2115

**Reply by chogberg** [reply]
I have not spent much time on this since I kind of lost steam because of this issue. If this is the case I think it is quite unhelpful to a noob like me to not be explicit that something is broken and display kind of a dryrun. I hope that bug get looked at by someone. Thanks for your input!
